chain/client/src/client.rs

```diff
@@ -1085,21 +1082,58 @@ impl Client {
         );
 
         // Send message to network to actually forward transaction.
-        self.network_adapter.do_send(NetworkRequests::ForwardTx(validator, tx));
+        self.network_adapter.do_send(NetworkRequests::ForwardTx(validator, tx.clone()));
 
-        NetworkClientResponses::RequestRouted
+        Ok(())
```

Comment:
Why not return `NetworkClientResponses` here?

Reply:
because we can only return NetworkClientResponses::RequestRouted

Reply:
Why is that a reason for not returning it? It removes the need to return `NetworkClientResponses` from the caller.

Reply:
@bowenwang1996 I'm with Misha here. If the outcome of the function is always the same except for errors, it should be `Result<(), Error>`.